CRM/SMS/Page/Callback.php

```diff
@@ -17,7 +17,13 @@
 class CRM_SMS_Page_Callback {
 
   public function run() {
-    $provider = CRM_SMS_Provider::singleton($_REQUEST);
+    try {
+      $provider = CRM_SMS_Provider::singleton($_REQUEST);
+    }
+    catch (CRM_Core_exception $e) {
```

Comment:
Also here - we are adding handling for what appears to be an obscure edge case bug where the no-change would be to leave it uncaught